### PR TITLE
enable tls access and upgrade generated code

### PR DESCRIPTION
**Pull request by elmiko**
This change enables https access, to do this the generated code has been
updated as has the api definition. This change also required the
upgrading of the graceful lib in the vendor directory.

Changes
* add https scheme to openapi definition
* upgrade generated code, using go-swagger version 0.5.0-180-g1693567
* upgrade graceful vendored library to current release

@crobby @sub-mod i have tested this locally with generated certs

### CODE REVIEW

**Comment by crobby** [comment]
Seems like a good start.  But without changes to start scripts and some sort of cert generation/documentation, I don't think it's quite ready to merge.  Of course, the re-generated bits might be more urgent than the tls part.  If that's the case, then I'm not against moving forward with that and having another PR for a more proper working tls setup.

**Comment by elmiko** [comment]
maybe for this PR, i should adjust the startup script so that it continues to work with just http. at least that way we remain backward compatible, but we are to enable https. would that work?

**Comment by crobby** [comment]
I think that might be good.

**Comment by sub-mod** [comment]
@elmiko IMO we don't need this right now.Since my current idea is use to tls in existing Routes.
Which sits well with my overall plan for security in Oshinko across webui/cli/rest.
```
apiVersion: v1
kind: Route
metadata:
 name: route-edge-secured 
spec:
 host: www.example.com
 to:
   kind: Service
   name: service-name 
 tls:
   termination: edge            
   key: |-                      
     -----BEGIN PRIVATE KEY-----
     [...]
     -----END PRIVATE KEY-----
   certificate: |-              
     -----BEGIN CERTIFICATE-----
     [...]
     -----END CERTIFICATE-----
```

We just can't rely only on secrets for certs in different containers.

**Comment by elmiko** [comment]
ok, i'm fine dropping this if we want. but, it should be completely backward compatible at this point, and provides us the option of using TLS at the server if we decide to.

**Comment by tmckayus** [comment]
I'm fine with this if it's backwards compatible.  It *is* possible to run oshinko-rest outside of a pod, I do it all the time for debugging.  Maybe we want an https option ...

**Comment by elmiko** [comment]
i'm thinking maybe i should update the readme as well? so there is some documentation about how to run from the command line with, and without, TLS support.

**Comment by tmckayus** [comment]
++ that seems like a good idea

File: README.md [hunk 1]
```diff
@@ -42,6 +42,27 @@ After building the binary a basic test can be performed as follows:
...
+file and a key file for the server to use. Once these files are created you
+can start the server as follows to enable HTTPS access:
+
+```
+    $ _output/oshinko-rest-server --port 42000 --tls-port 42443 --tls-key keyfile.key --tls-certificate certificatefile.cert
+    2016/09/28 12:10:47 Serving oshinko rest at http://127.0.0.1:42000
+    2016/09/28 12:10:47 Serving oshinko rest at https://127.0.0.1:42443
+```
+
+At this point the server is ready to accept both HTTP and HTTPS requests. If
```
Comment (crobby) [inline README.md]: Nit:  If *you* would like

Comment (elmiko) [inline README.md]: done

File: README.md [hunk 2]
```diff
@@ -55,8 +55,8 @@ can start the server as follows to enable HTTPS access:
 ```
 
 At this point the server is ready to accept both HTTP and HTTPS requests. If
-would like to resitrct access to **only** use TLS, add the `--scheme https`
-flag to the command line as follows:
+you would like to resitrct access to **only** use TLS, add the
```
Comment (elmiko) [inline README.md]: done

**Pull request merged**